nntrainer/tensor/tensor.cpp

```diff
@@ -251,36 +275,86 @@ Tensor Tensor::add(float const &value, Tensor &out) const {
   return out;
 }
 
-/**
- * @struct External Loop Info for broadcasted info
- * @brief External Loop Info for broadcasted iteration. Please refer to
- * DISABLED_private_external_loop_n in unittest_nntrainer_tensor.
- * @note This should better be implemented in iterator fashion before used
- * extensively.
- */
-struct Tensor::BroadcastInfo {
+int Tensor::subtract_i(float const &value) { return this->add_i(-value); }
 
-  /**
-   * @brief Construct a new External Loop Info object
-   *
-   */
-  BroadcastInfo() : strides{0, 0, 0, 0} {}
+Tensor Tensor::subtract(float const &value) const { return this->add(-value); }
 
-  unsigned int buffer_size; /**< virtual size of the buffer */
-  int buffer_axis;          /**< the smallest axis that should be looped.
-                                 -1 means no loop needed*/
-  std::array<unsigned int, MAXDIM>
-    strides; /**< modified strides for the loop */
-};
+Tensor Tensor::subtract(float const &value, Tensor &out) const {
+  throw nntrainer::exception::not_supported("NYI");
+}
+
+Tensor Tensor::pow(float exponent) const {
+  return apply([=](float in) { return powf(in, exponent); });
+}
+
+int Tensor::pow_i(float exponent) {
+  return apply_i([=](float in) { return powf(in, exponent); });
+}
+
+int Tensor::multiply_i(Tensor const &m) {
```

Comment:
as this PR is doing re-arranging, let club all the forms of multiply together, and same for other operations too.

Reply:
I found that multiply with tensor has more common characteristic with divide with tensor then multiply with scalar hence this arrangement. If we are to add another arithmetic operator (e.g sparse tensor???), I think those could be easily cluster as well.

Reply:
It will be best if the arrangement in the header and source files are synced.
From a code reviewer/reader perspective, who wants to find a function, say multiply, it's difficult to find a required function at different sections of the files compared to the arrangement where all the different interfaces of multiply listed together.

Reply:
Sounds reasonable, I'll change :)

Reply:
Done